utils/hare-shutdown

```diff
@@ -17,217 +17,78 @@
-
-def exec_silent(cmd: str) -> bool:
-    return os.system(cmd) == 0
-
-
-def exec(cmd: str) -> None:
-    assert cmd
-    if exec_silent(cmd):
-        logging.info('OK')
-    else:
-        logging.error('**ERROR**')
-
-
-def process_stop(proc: Process) -> None:
-    if proc.status != 'passing':
-        return
-    label = f' ({proc.consul_name})' if proc.systemd_name.startswith('m0d@') \
-        else ''
-    logging.info(f'Stopping {proc.systemd_name}{label} at {proc.node}... ')
-    ok = exec_silent('{}sudo systemctl stop --force {}'.format(
-        ssh_prefix(proc.node), proc.systemd_name))
-    if ok:
-        logging.info(f'Stopped {proc.systemd_name}{label} at {proc.node}')
-    else:
-        logging.error(f'**ERROR** Failed to stop {proc.systemd_name}{label}'
-                      f' at {proc.node}')
 
-
-class StopProcess:
-    def __init__(self, process: Process):
-        self.process = process
-
-
-class QuitMessage:
-    pass
-
-
-class Worker(threading.Thread):
-    def __init__(self, queue: Queue):
-        super().__init__(target=self._do_work, args=(queue, ))
-
-    def _do_work(self, queue):
-        logging.debug('Started thread')
-        while True:
-            msg = queue.get()
-            if isinstance(msg, StopProcess):
-                process_stop(msg.process)
-            if isinstance(msg, QuitMessage):
-                logging.debug('Exiting thread')
-                return
-
-
-def _stop_parallel(process_list: List[Process], thread_count: int = 8) -> None:
-    size = len(process_list)
-
-    if not size:
-        return
-    q: Queue = Queue(maxsize=size)
-    thread_count = min(thread_count, size)
-
-    worker_pool: List[Worker] = [Worker(q) for i in range(thread_count)]
-
-    for w in worker_pool:
-        w.start()
-
-    for proc in process_list:
-        q.put(StopProcess(process=proc))
-    for i in range(thread_count):
-        q.put(QuitMessage())
-
-    for worker in worker_pool:
-        worker.join()
-
-
-def main() -> None:
-    _setup_logging()
-    if not consul_is_active_at('localhost'):
-        if pcs_consul_is_active_at('localhost'):
+def check_cluster():
+    if not utils.consul_is_active_at('localhost'):
+        if utils.pcs_consul_is_active_at('localhost'):
             exit("Cluster is run by Pacemaker, try `pcs cluster stop --all'")
         else:
             exit('Cluster is not running')
 
+
+@repeat_if_fails()
+def _get_processes_and_node():
     cns = Consul()
+    processes = utils.processes_by_consul_svc_name(cns)
+    leader_node = utils.get_kv(cns, 'leader')
+    return (processes, leader_node)
 
-    @repeat_if_fails()
-    def _get_processes_and_node():
-        processes = processes_by_consul_svc_name(cns)
-        leader_node = get_kv(cns, 'leader')
-        return (processes, leader_node)
 
+def shutdown_cluster():
+    check_cluster()
     (processes, leader_node) = _get_processes_and_node()
-    for svc in shutdown_sequence:
+    for svc in utils.shutdown_sequence:
```

Comment:
this loop will go to `node_shutdown()`, here you need to iterate over `processes` and pass each processes to `node_shutdown()`.